File: Data Structures & Algorithms/binary-search/submission-0.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int l = 0, r = nums.size()-1;
        int mid;

        while(l <= r){
            mid = (l+r)/2;
            cout<<nums[mid]<<"\n";
            if(nums[mid] == target) return mid;
            else if(nums[mid] < target) l = mid+1;
            else r = mid-1;
        }
        return -1;
    }
};

File: Data Structures & Algorithms/combination-target-sum/submission-0.py
from typing import List

class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        res = []
        count = 0
        arr = []
        sum = 0
        def dfs(i):
            
            nonlocal sum
            print(arr)
            
            if i >= len(nums):
                return
            
            # Base Case
            if sum > target:
                return
                
            # Found a result
            if sum == target:
                res.append(arr.copy())
                return 
                
            arr.append(nums[i])
            sum += nums[i]
            dfs(i)
            
            arr.pop()
            sum -= nums[i]
            dfs(i+1)
        
        dfs(0)
            
        return res